server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -223,9 +225,20 @@ func (s *BuildBuddyServer) CreateGroup(ctx context.Context, req *grpb.CreateGrou
 		OwnedDomain: groupOwnedDomain,
 	}
 	urlIdentifier := strings.TrimSpace(req.GetUrlIdentifier())
+
+	// Make sure the URL identifier isn't already taken by any existing group.
 	if urlIdentifier != "" {
+		existingGroup, err := userDB.GetGroupByURLIdentifier(ctx, urlIdentifier)
```

Comment:
if existingGroup, err := userDB.GetGroupByURLIdentifier(ctx, urlIdentifier); existingGroup != nil {
  return status.InvalidArgumentError("URL is already in use")
} else {
  if err != notfound {
    return err
  }
}

how about this? just trying to map the code as closely as possible to the logic of like... if grp is found, return in use, else if other unexpected err, return it.

Reply:
Nice, much cleaner. Done!